rfcs/0093-rfc-categories.md

```diff
@@ -0,0 +1,152 @@
+---
+feature: rfc_categories
+start-date: 2021-05-19
+author: David Arnold
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @kevnicox, @Mic92, @gytis-ivaskevicius
+shepherd-leader: @kevincox
+related-issues:
+category: process
+---
+
+# Summary
+[summary]: #summary
+
+The RFC process is amended with means to categorize RFCs into one of: _feature_, 
+_information_ or _process_, where each category sets different accents that
+improve the overall outcome of the process.
+
+# Motivation
+[motivation]: #motivation
+
+## Some issues are not addressed by the community
+
+There is no appropriate venue for it. When reading the current RFC process, it becomes
+clear that it was made primarily with purely technical decisions in mind. It might only
+be a perception, but this perception systemically skews what's being discussed and what
+not.
+
+As a result, important topics might not be adressed and settled by the community at large
+in a generally accepted procedure.
+
+Specifically, it is hard to propose a well coordinated experiment across the community,
+document and acknowledge design issues, record proof-generated insight, but also to
+amend the RFC process itself, the forum rules, the code of conduct or propose any other 
+binding changes to community workflows or infrastructure.
+
+**For example**, the [flake RFC](https://github.com/NixOS/rfcs/pull/49), which by many is
+considered a failed RFC process, might have benefited from a framework to transition into
+a general experiment in the form of an _informational_ RFC as soon as it had become
```

Comment:
If flakes would've been an informational RFC, then I think the word informational is a misnomer.

Reply:
It might be. I'd be happy to hear suggestions.

A better name for something that does not primarily originate in a change.

Let's keep in mind that everything can be technically interpreted as change, but contrasting with the other two categories, this category attempts to create a "space for RFC-style consensus" that does not _primarily_ originate in a change.

Reply:
I think the flake RFC had similar problems, but maybe wouldn't be solved exactly by this proposal.

I think it *was* a problem that we couldn't merge an RFC with the intent to experiment. This is an example of an issue specifying the desired outcome. It is very common for people to focus on "I want to make X easier" without saying the "implied" bit of "I want to **make a technical change** to Nix that makes X easier". For example maybe if we could add "I want to **add an experimental feature to nix** to make X easier. This could have guided reviewers that not every technical detail needs to be perfect and production read, as another RFC would be required to actually implement the stable feature once we have a lot more background. This could have helped the RFC be accepted with the right goals.

This sounds similar to this RFC where categories are a way of suggesting the desired outcome. _feature_ implies that you want to make a change to Nix, Nixpkgs or whatever and _informational_ means that you just want to get the community to agree on a fact or opinion with no other change. However the difference here is that flakes are not _informational_. The intention is to make a change to Nix, but maybe it would have helped to make a *not officially supported* change which is the key difference missing from the objective.

I do see a couple of ways to handle this:
- Recommend that some or all technical features have an experiment first. (A two-step process by default). 
- Add an experimental category to help direct the review.

However I don't see this RFC doing either. So I think the example needs to be changed to not be putting the flakes RFC into a category in which it doesn't appear to fit. Either we need to find a category for it that would be expected to help the review process or we need to find a better example.